dora/core/server/master/src/main/java/alluxio/master/job/CopyJob.java

```diff
@@ -302,9 +302,14 @@ public List<CopyTask> getNextTasks(Collection<WorkerInfo> workers) {
       if (routes.isEmpty()) {
         return Collections.unmodifiableList(tasks);
       }
-      WorkerInfo workerInfo = mWorkerAssignPolicy.pickAWorker(StringUtil.EMPTY_STRING, workers);
+      List<WorkerInfo>
+          workerInfo = mWorkerAssignPolicy.pickWorkers(StringUtil.EMPTY_STRING, workers, 1);
+      if (workerInfo.isEmpty()) {
+        LOG.warn("No worker available for copy job {}.", mJobId);
+        mRetryRoutes.addAll(routes);
```

Comment:
bail here? otherwise workerInfo.get(0) has index out of bound

Reply:
fixed